### PR TITLE
trying to fix integrity errors before migrating tables

### LINKED ISSUE #12138: Handle Integrity Error in Django 3.2 Upgrade
## Observed behavior
Encountering integrity errors post-upgrade to Django 3.2 due to invalid data in the local database. Not sure how we got here but worth looking into.

## Errors and logs

```
[python-devserver] django.db.utils.IntegrityError: 
The row in table 'logger_usersessionlog' with primary key '905c3e6ef149e1b4bacddfe782ff906f' 
has an invalid foreign key: logger_usersessionlog.user_id 
contains a value '2f4536317ea71e134e84f0fb39698c02' 
that does not have a corresponding value in kolibriauth_facilityuser.id.
```

## Steps to reproduce
Upgrade to Django 3.2 
Reinstall  Python dependencies

## Context
develop

…

### PR DESCRIPTION
**Pull request by jredrejo**
## Summary
Before updating kolibri database on a Django upgrade, check integrity of the tables and fix automatically errors that can be safely fixed

## References
Closes: #12138

## Reviewer guidance
Using a broken db, check that it's fixed.
Using a good db, check no problems are added

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/utils/main.py [hunk 1]
```diff
@@ -185,6 +186,27 @@ def _copy_preseeded_db(db_name, target=None):
             )
 
 
+def sqlite_check_foreign_keys():
+    db_connection = connections["default"]
```
Comment (rtibbles) [inline kolibri/utils/main.py]: This is explicitly being done in the method that is before Django setup, so I think using the django DB connection here might not be a good idea. Could you instead generate the connection directly using the Python sqlite library?

We should also iterate over all the SQLite  databases listed in https://github.com/learningequality/kolibri/blob/develop/kolibri/deployment/default/sqlite_db_names.py#L14 to be safe.

**Review by rtibbles — CHANGES_REQUESTED** [review]
One tweak to avoid premature Django initialization, which I also think will help with the testing error.

File: kolibri/utils/main.py (hunk 1 above)
Comment (jredrejo) [reply kolibri/utils/main.py]: argg, got it. Thanks for the advice
Changed as requested

**Review by rtibbles — COMMENTED** [review]
Code looks right, I'll test it with the database that @AllanOXDi had with the original issue. Thanks!

**Review by rtibbles — APPROVED** [review]
Manual testing with an affected database shows that the fix is applied and logged, and manual inspection of the database afterwards shows that the affected rows have been removed.

**Pull request merged**